client/src/components/ui/Modal.svelte

```diff
@@ -0,0 +1,83 @@
+<script lang="ts">
+    import Button from "./Button.svelte";
+
+    export let showModal = false;
+    export let isForm = false;
+
+    let dialog: HTMLDialogElement;
+
+    $: if (dialog && showModal) dialog.showModal();
```

Comment:
Technically, the `dialog` will not be initialized until the [`onMount`](https://svelte.dev/docs#run-time-svelte-onmount). Until then, the variable is actually `undefined`.

To enforce type safety, we initialize the `dialog` to `null`.

```suggestion
    let dialog: HTMLDialogElement | null = null;

    $: if (showModal) dialog?.showModal();
```